From daded1aff280104d16e405fcd1be1a857c74b191 Mon Sep 17 00:00:00 2001
From: Kamil Dudka <kdudka@redhat.com>
Date: Mon, 27 Aug 2018 15:53:35 +0200
Subject: [PATCH] tests: make ssh-keygen always produce PEM format

The default format produced by openssh-7.8p1 cannot be consumed
by currently available versions of libssh and libssh2.
---
 tests/sshserver.pl | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/sshserver.pl b/tests/sshserver.pl
index 9b3d122..d477a02 100755
--- a/tests/sshserver.pl
+++ b/tests/sshserver.pl
@@ -372,12 +372,12 @@ if((! -e $hstprvkeyf) || (! -s $hstprvkeyf) ||
     # Make sure all files are gone so ssh-keygen doesn't complain
     unlink($hstprvkeyf, $hstpubkeyf, $cliprvkeyf, $clipubkeyf);
     logmsg 'generating host keys...' if($verbose);
-    if(system "\"$sshkeygen\" -q -t rsa -f $hstprvkeyf -C 'curl test server' -N ''") {
+    if(system "\"$sshkeygen\" -q -t rsa -f $hstprvkeyf -C 'curl test server' -N '' -m PEM") {
         logmsg 'Could not generate host key';
         exit 1;
     }
     logmsg 'generating client keys...' if($verbose);
-    if(system "\"$sshkeygen\" -q -t rsa -f $cliprvkeyf -C 'curl test client' -N ''") {
+    if(system "\"$sshkeygen\" -q -t rsa -f $cliprvkeyf -C 'curl test client' -N '' -m PEM") {
         logmsg 'Could not generate client key';
         exit 1;
     }
-- 
2.17.1

